### ISSUE TITLE
[cmake] Add AVX config flag for other platforms, in line with usage for windows

### ISSUE THREAD

**Issue by kunaltyagi** [PointCloudLibrary/pcl#4693]
I'm not that well versed in Linux / cross compiling so I would prefer to just merge this one and one with better knowledge of linux can adjust it accordingly. Or someone make PR towards this PR.

_Originally posted by @larshg in https://github.com/PointCloudLibrary/pcl/issues/4598#issuecomment-812146560_

**Comment by ueqri** [comment]
I can handle the issue to add AVX or AVX2 config for Linux and macos in CMakeLists.

Could I make another PR after #4598 being merged? Because I have no permission to contribute new codes in that PR at present. :joy:

**Comment by kunaltyagi** [comment]
> Could I make another PR after #4598 being merged?

Yes, that'd be a good way to go

**Comment by mvieth** [comment]
Added by pull request #4698
